airbyte-cdk/python/unit_tests/sources/file_based/in_memory_files_source.py

```diff
@@ -0,0 +1,68 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import csv
+import io
+from datetime import datetime
+from io import IOBase
+from typing import Dict, List, Optional, Type
+
+from airbyte_cdk.sources.file_based.discovery_policy import AbstractDiscoveryPolicy
+from airbyte_cdk.sources.file_based.file_based_source import FileBasedSource
+from airbyte_cdk.sources.file_based.file_based_stream_reader import AbstractFileBasedStreamReader
+from airbyte_cdk.sources.file_based.file_types.file_type_parser import FileTypeParser
+from airbyte_cdk.sources.file_based.remote_file import RemoteFile
+from airbyte_cdk.sources.file_based.stream import AbstractFileBasedStream
+from airbyte_cdk.sources.streams.availability_strategy import AvailabilityStrategy
+
+
+class InMemoryFilesSource(FileBasedSource):
+    def __init__(
+            self,
+            files,
+            file_type,
+            availability_strategy: AvailabilityStrategy,
+            discovery_policy: AbstractDiscoveryPolicy,
+            parsers: Dict[str, FileTypeParser],
+            stream_cls: Type[AbstractFileBasedStream],
+    ):
+        super().__init__(
+            InMemoryFilesStreamReader(files=files, file_type=file_type),
+            availability_strategy=availability_strategy,
+            discovery_policy=discovery_policy,
+            parsers=parsers,
+        )
+        self.files = files
+        self.file_type = file_type
+        self.discovery_policy = discovery_policy
+
+
+class InMemoryFilesStreamReader(AbstractFileBasedStreamReader):
+    files: Dict[str, dict]
+    file_type: str
+
+    def list_matching_files(
+        self,
+        globs: List[str],
+        from_date: Optional[datetime] = None,
+    ) -> List[RemoteFile]:
+        return [
+            RemoteFile(f, data["last_modified"], self.file_type)
```

Comment:
nit: [data["last_modified"] is a string ](https://github.com/airbytehq/airbyte/pull/27122/files#diff-261d057402b704966dbd561ac054ecb582646d2e5cdef5ee57ef81d6bfc32e62R32) so we should parse it (datetime.strptime(`data["last_modified"], "%Y-%m-%dT%H:%M:%S.%fZ"`)